packages/apollo-language-server/src/project.ts

```diff
@@ -201,27 +200,44 @@ export class GraphQLProject {
               })
             );
 
+            type FieldStat = {
+              groupBy: {
+                field: string;
+              };
+              metrics: {
+                fieldHistogram: {
+                  durationMs: number;
+                };
+              };
+            };
+
             const schemaEngineStats = new Map<string, Map<string, number>>();
-            engineData.data!.service.report.usageStats.types.forEach(
-              (typ: any) => {
-                const fieldsMap = new Map<string, number>();
-                typ.fields.forEach((field: any) => {
-                  fieldsMap.set(
-                    field.name,
-                    field.latencyHistogram.serviceTimeP95
-                  );
-                });
-
-                schemaEngineStats.set(typ.name, fieldsMap);
+            engineData.data!.service.stats.fieldStats.forEach(
+              (fieldStat: FieldStat) => {
+                // Parse field "ParentType.fieldName:FieldType" into ["ParentType", "fieldName", "FieldType"]
```

Comment:
Whats going on with this? How does this data come back? Is this a change we should add to the server?

Reply:
Good question! I just had some discussion with Danielle and @evans about this and learned quite a bit. The shape of this query relates directly to Druid and the utility we use that generates resolvers for druid queries. So...I do agree that parsing this on the frontend isn't ideal but I don't think it justifies a change to the resolvers. Did I understand your question correctly?

Reply:
@jbaxleyiii I'm looking to merge soon, can I resolve this or did you want to dig any deeper?